site/en/docs/extensions/mv3/service_workers/events/index.md

```diff
@@ -80,10 +81,12 @@ Extension service workers support more than the lifecycle events [described else
 
 ### ServiceWorkerGlobal.fetch {: #fetch }
 
-Fired when anything is retrieved from the extension package or when `fetch()` and `XMLHttpRequest()` are called from a content or popup script. In the latter cases, you will need to add the URLs of the pages you want to fetch to the `"host_permissions"` key in the `manifest.json`.
+Fired when anything is retrieved from the extension package or when `fetch()` and `XMLHttpRequest()` are called from an extension or popup script. In the latter cases, you will need to add the URLs of the pages you want to fetch to the `"host_permissions"` key in the `manifest.json`.
+
+There are a few things to note about this. First, content scripts cannot call `fetch()` since they run in a host page's origin. Also, only use the `fetch()` method for calling web services, static resources, or any use case that doesn't download code. Downloaded code is [not allowed in Manifest V3](/docs/extensions/migrating/improve-security/#remove-remote-code). 
```

Comment:
```suggestion
`fetch` events are rarely used in Extensions and they are only listed here for reference purposes. In general, there are better ways to accomplish the same thing in an extension.
```

This is not quite correct, fetch is supported in content scripts, but they won't fire a fetch event. The warning about downloading code is not relevant in this context.

Reply:
It appears that the ball is in your court.